odaridavid/POLITICO-API
# #163893944 Add Update Endpoint on Offices

### What Does This PR do?
- Add Route to Update on Office Endpoint with the  PATCH request method
- Made a general method to deal with getting a specific name for edit with
 offices name and party name
 - Add Tests for office name updating and refactored existing tests for party
- Update Readme with Office Update Endpoint

### Description of Task To Be Completed?

Have the `PATCH` Method Request on the `/offices` Endpoint working for updating an office  and returning a JSON response of the name and id of office changed

### How Can This Be Manually Tested 
1. Clone The Repository and git checkout to ft-update-office-163893944.
2. Start Flask Virtual environment and export configuration details to run an instance of flask app.
3. On Postman or Similar API, Testing Client. Use the `POST` Method to add a office in the format below to the `/office` endpoint.
```
{
	"name":"office_name",
	"type":"office_type"
}
```
4. Check Id generated and use the `PATCH` method on the `/offices/(generated_id)/name` it should return a 200 Response code and new item name 
5. To check it reflected on the list use the `GET` Request Method and view the item on generated id

### Relevant PT Stories

_#163893944_

### Screenshots 

![screenshot from 2019-02-12 07-56-25](https://user-images.githubusercontent.com/20339228/52612891-adf80800-2e9c-11e9-80f6-b8a740837c1c.png)
![screenshot from 2019-02-12 07-56-31](https://user-images.githubusercontent.com/20339228/52612893-adf80800-2e9c-11e9-87cc-da63c71d1c0b.png)
